hybridauth.module

```diff
@@ -12,8 +12,235 @@ function hybridauth_form_alter(&$form, FormStateInterface $form_state, $form_id)
     $form['hybridauth'] = [
       '#title' => t('Log in with Linkedin'),
       '#type' => 'link',
-      '#url' => Url::fromRoute('hybridauth.authenticate', ['provider_id' => 'linkedin']),
+      '#url' => Url::fromRoute('hybridauth.authenticate', ['provider_id' => 'LinkedIn']),
     ];
+
+    $form['hybridauth_facebook'] = [
+      '#title' => t('Log in with Facebook'),
+      '#type' => 'link',
+      '#url' => Url::fromRoute('hybridauth.authenticate', ['provider_id' => 'Facebook']),
+    ];
+
+    $form['hybridauth_google'] = [
+      '#title' => t('Log in with Google'),
+      '#type' => 'link',
+      '#url' => Url::fromRoute('hybridauth.authenticate', ['provider_id' => 'Google']),
+    ];
+
+    $form['hybridauth_bitbucket'] = [
+      '#title' => t('Log in with BitBucket'),
+      '#type' => 'link',
+      '#url' => Url::fromRoute('hybridauth.authenticate', ['provider_id' => 'BitBucket']),
+    ];
+
+    $form['hybridauth_yahoo'] = [
+      '#title' => t('Log in with Yahoo'),
+      '#type' => 'link',
+      '#url' => Url::fromRoute('hybridauth.authenticate', ['provider_id' => 'Yahoo']),
+    ];
+  }
+
+}
+
+/**
+ * Returns auth provider name by provider ID.
+ */
+function hybridauth_get_provider_name($provider_id) {
+  $providers = hybridauth_providers_list();
+  return isset($providers[$provider_id]) ? $providers[$provider_id] : NULL;
+}
+
+/**
+ * Internal functions.
+ */
+function hybridauth_providers_list($reset = FALSE) {
+  $providers = &drupal_static(__FUNCTION__, NULL);
+
+  if (!isset($providers)) {
+    $raw_providers = [
```

Comment:
this list should be automatically generated based on hybridauth providers list (see https://github.com/hybridauth/hybridauth/tree/master/src/Provider). No replication anymore. Take label from the class name

Reply:
Thank you. Changed.